kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/KuraModemGPSMode.java

```diff
@@ -0,0 +1,71 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Eurotech and/or its affiliates and others
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *  Eurotech
+ ******************************************************************************/
+package org.eclipse.kura.nm;
+
+import java.util.EnumSet;
+import java.util.Set;
+
+import org.eclipse.kura.nm.enums.MMModemLocationSource;
+
+public enum KuraModemGPSMode {
+
+    KURA_MODEM_GPS_MODE_UNMANAGED("kuraModemGpsModeUnmanaged"),
+    KURA_MODEM_GPS_MODE_MANAGED_GPS("kuraModemGpsModeManagedGps");
+
```

Comment:
Does it make sense to call it `KURA_MODEM_GPS_MODE_MANAGED`?

Reply:
Legitimate question.

I decided to call it this way since ModemManager also supports the ["Assisted-GPS" mode](https://www.freedesktop.org/software/ModemManager/doc/latest/ModemManager/ModemManager-Flags-and-Enumerations.html#MMModemLocationSource) (i.e. `MM_MODEM_LOCATION_SOURCE_AGPS_MSA` and `MM_MODEM_LOCATION_SOURCE_AGPS_MSB`).

If, in the future, we want to add the support for this additional mode there's no clash with the existing name and we can simply add something like `KURA_MODEM_GPS_MODE_MANAGED_AGPS_MSA` and `KURA_MODEM_GPS_MODE_MANAGED_AGPS_MSB`.